src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,43 @@
+package duke.parser;
+
+import duke.command.OtherCommand;
+import duke.command.MarkingCommand;
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.ExitCommand;
+import duke.command.DeleteCommand;
+import duke.dukeexception.DukeException;
+
+/**
+ * Represents a function of Duke robot, which can produce command corresponding to user input.
+ */
+public class Parser {
+    /**
+     * Produce different type of command for execution, corresponding to user input.
+     * Throws DukeException when the user input format is not right.
+     * @param fullCommand A string of a line from System.in. Can be of any format.
+     * @return A certain kind of command waiting for execution.
+     * @throws DukeException Throws DukeException with remind message when the input format is wrong.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```

Comment:
Quite inefficient to keep splitting the string until you get the correct command. Can consider just doing it once storing it as a variable then checking to see if the word is a command.